File: data/src/main/java/org/cryptomator/data/cloud/dropbox/RootDropboxFolder.java
package org.cryptomator.data.cloud.dropbox;

import org.cryptomator.domain.Cloud;
import org.cryptomator.domain.DropboxCloud;

class RootDropboxFolder extends DropboxFolder {

	private final DropboxCloud cloud;

	public RootDropboxFolder(DropboxCloud cloud) {
		super(null, "", "");
		this.cloud = cloud;
	}

	@Override
	public DropboxCloud getCloud() {
		return cloud;
	}

	@Override
	public DropboxFolder withCloud(Cloud cloud) {
		return new RootDropboxFolder((DropboxCloud) cloud);
	}
}

File: data/src/main/java/org/cryptomator/data/cloud/dropbox/RootDropboxFolder.kt
package org.cryptomator.data.cloud.dropbox

import org.cryptomator.domain.Cloud
import org.cryptomator.domain.DropboxCloud

internal class RootDropboxFolder(override val cloud: DropboxCloud) : DropboxFolder(null, "", "") {

	override fun withCloud(cloud: Cloud?): DropboxFolder {
		return RootDropboxFolder(cloud as DropboxCloud)
	}
}
